### PR TITLE
add api-gateway

### PR DESCRIPTION
**Pull request by akinoniku**
Only and must have one lambda "service", can use api-gatway

TODO: custom domain & cert if needed

### CODE REVIEW

File: vars.tf [hunk 1]
```diff
@@ -1,3 +1,13 @@
+variable "aws_region" {
```
Comment (rdunn-Hypr) [inline vars.tf]: The region is not a concern of the resource module. Please remove

File: vars.tf [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+  type        = string
+  description = "aws-region: e.g ap-southeast-2"
+}
+
+variable "aws_account_number" {
```
Comment (rdunn-Hypr) [inline vars.tf]: the account number is not a concern of the resource module please remove it

File: vars.tf [hunk 3]
```diff
@@ -332,4 +342,14 @@ variable "s3_tags" {
   default     = {}
 }
 
+variable "enable_api_gateway" {
```
Comment (rdunn-Hypr) [inline vars.tf:366]: The architecture has a language of internal and external endpoints, of which external endpoints could have a type - S3, API gateway - we're still reasoning about these but I'm unsure if I want an explicit flag that controls adding an api-gateway resource

File: vars.tf [hunk 4]
```diff
@@ -332,4 +342,14 @@ variable "s3_tags" {
...
+  type        = bool
+  description = "allow to create api-gateway"
+  default     = false
+}
+
+variable "api_gateway_name" {
```
Comment (rdunn-Hypr) [inline vars.tf]: Should be computed base of the name for the application, look at the event bus as an example

File: api-gateway.tf [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  target        = aws_lambda_function.lambda_application["service"].arn
+  count         = var.enable_api_gateway ? 1 : 0
+}
+
+resource "aws_lambda_permission" "allow_apigateway" {
+  statement_id  = "AllowExecutionFromApiGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.lambda_application["service"].function_name
+  principal     = "apigateway.amazonaws.com"
+  source_arn    = "arn:aws:execute-api:${var.aws_region}:${var.aws_account_number}:${aws_apigatewayv2_api.api-gateway[0].id}/*"
```
Comment (rdunn-Hypr) [inline api-gateway.tf]: is this not the arn of the `aws_apigatewayv2_api.api-gateway` resource?

**Review by rdunn-Hypr — CHANGES_REQUESTED** [review]
There are still some gaps in your understanding of the intent of the module design. I am a little distracted with some other work, so it's quite difficult to give you the support you need

File: vars.tf (hunk 3 above)
Comment (akinoniku) [reply vars.tf:366]: Does that mean the module should not define external endpoints?

File: vars.tf (hunk 1 above)
Comment (akinoniku) [reply vars.tf]: removed

File: vars.tf (hunk 4 above)
Comment (akinoniku) [reply vars.tf]: fixed

File: api-gateway.tf (hunk 5 above)
Comment (akinoniku) [reply api-gateway.tf]: fixed

File: vars.tf (hunk 3 above)
Comment (rdunn-Hypr) [reply vars.tf:366]: I feel the module takes a list or dictionary of external endpoints. these external endpoints would contain a type, e.g. api-gateway, S3 etc

What I'm unsure about is if there are that many external endpoints, in that case, saying api_gateway_enabled could make sense

Comment (akinoniku) [reply vars.tf:366]: It could be, I talked to team who. which they might want to use SQS as an external endpoint.

**Review by rdunn-Hypr — APPROVED** [review]
Thanks for making those changes. This is likely to evolve next sprint

**Comment by rdunn-Hypr** [comment]
Code changes in this PR have been merged in PR #3 Am closing this PR

**Pull request closed without merging**